File: tareas/src/main/resources/static/js/app/TaskService.js
'use strict'
angular.module('tasks.services', []).factory('TaskService', ["$http", "CONSTANTS", function($http, CONSTANTS) {
    var service = {};
    service.getTaskById = function(taskId) {
        var url = CONSTANTS.getTaskById + taskId;
        return $http.get(url);
    }
    service.getAllTasks = function() {
        return $http.get(CONSTANTS.getAllTasks);
    }
    service.saveTask = function(taskDto) {
        return $http.post(CONSTANTS.saveTask, taskDto);
    }
    service.updateTask = function(taskDto) {
        return $http.post(CONSTANTS.updateTask, taskDto);
    }
    service.deleteTask = function(taskId) {
        return $http.post(CONSTANTS.deleteTask, taskId);
    }
    return service;
}]);

File: tareas/src/main/java/com/tasks/model/service/TaskService.java
package com.tasks.model.service;

import com.tasks.model.dto.TaskDto;

import java.util.List;

public interface TaskService {

    /**
     * Devolvemos tarea por id
     * @param userId
     * @return
     */
    TaskDto getTaskById(Integer userId);

    /**
     * Guarda tarea
     * @param userDto
     */
    void saveTask(TaskDto userDto);

    /**
     * Actualiza tarea
     * @param userDto
     */
    void updateTask(TaskDto userDto);

    /**
     * Devuelve todas las tareas
     * @return
     */
    List< TaskDto > getAllTasks();

    /**
     * Borra tarea
     * @param userId
     */
    void deleteTask(Integer userId);
}
